wfe/web-front-end.go

```diff
@@ -757,46 +760,93 @@ func (wfe *WebFrontEndImpl) NewCertificate(response http.ResponseWriter, request
 	wfe.Stats.Inc("Certificates", 1, 1.0)
 }
 
-func (wfe *WebFrontEndImpl) challenge(
+func (wfe *WebFrontEndImpl) Challenge(
 	response http.ResponseWriter,
-	request *http.Request,
-	authz core.Authorization,
-	logEvent *requestEvent) {
+	request *http.Request) {
+	logEvent := wfe.populateRequestEvent(request)
+	defer wfe.logRequestDetails(&logEvent)
 
-	// Check that the requested challenge exists within the authorization
-	found := false
-	var challengeIndex int
-	var challenge core.Challenge
-	for i, challenge := range authz.Challenges {
-		tempURL := challenge.URI
-		if tempURL.Path == request.URL.Path && tempURL.RawQuery == request.URL.RawQuery {
-			found = true
-			challengeIndex = i
-			break
-		}
+	notFound := func() {
+		wfe.sendError(response, "No such registration", request.URL.Path, http.StatusNotFound)
 	}
 
-	if !found {
-		logEvent.Error = "Unable to find challenge"
-		wfe.sendError(response, logEvent.Error, request.URL.RawQuery, http.StatusNotFound)
+	// Challenge URIs are of the form /acme/challenge/<auth id>/<challenge id>.
+	// Here we parse out the id components. TODO: Use a better tool to parse out
+	// URL structure: https://github.com/letsencrypt/boulder/issues/437
+	slug := strings.Split(request.URL.Path[len(ChallengePath):], "/")
+	if len(slug) != 2 {
+		notFound()
 		return
 	}
+	authorizationID := slug[0]
+	challengeID, err := strconv.ParseInt(slug[1], 10, 64)
```

Comment:
ah true, avoiding the cast is probably worth it then.

On Wed, Sep 9, 2015 at 4:50 PM Jacob Hoffman-Andrews <
notifications@github.com> wrote:

> In wfe/web-front-end.go
> <https://github.com/letsencrypt/boulder/pull/764#discussion_r39111393>:
>
> >  		return
> >  	}
> > +	authorizationID := slug[0]
> > +	challengeID, err := strconv.ParseInt(slug[1], 10, 64)
>
> Unfortunately Atoi returns int, and this needs to be int64.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/letsencrypt/boulder/pull/764/files#r39111393>.
>